File: 17-Chain-of-Responsibility/src/go/team_lead.go
package main

import "fmt"

type TeamLead struct {
	BaseApprover
}

func (t *TeamLead) Handle(request PurchaseRequest) {
	if request.Amount <= 1000 {
		fmt.Printf("  Team Lead approved $%.2f for %s\n", request.Amount, request.Purpose)
	} else {
		t.passToNext(request)
	}
}

File: 17-Chain-of-Responsibility/src/java/chapter17/TeamLead.java
package chapter17;

public class TeamLead extends Approver {
    public TeamLead() {
        super("Team Lead", 1000);
    }

    @Override
    protected void approve(PurchaseRequest request) {
        System.out.printf("  %s approved $%.2f for %s%n",
                name, request.getAmount(), request.getPurpose());
    }
}
